hedera-node/src/main/java/com/hedera/services/ledger/TransactionalLedger.java

```diff
@@ -77,25 +82,29 @@
 	private final TransactionalLedger<K, P, A> entitiesLedger;
 	private final ChangeSummaryManager<A, P> changeManager;
 	private final Function<K, EnumMap<P, Object>> changeFactory;
+	private final SideEffectsTracker sideEffectsTracker;
 
 	final Map<K, EnumMap<P, Object>> changes = new HashMap<>();
 
 	private boolean isInTransaction = false;
-	private PropertyChangeObserver<K, P> commitInterceptor = null;
+	private PropertyChangeObserver<K, P> propertyChangeObserver = null;
 	private Optional<Function<K, String>> keyToString = Optional.empty();
+	private final CommitInterceptor<K, A, P> commitInterceptor = this::checkHbarChanges;
```

Comment:
I think we'll want to define the `CommitInterceptor` implementations in stand-alone classes; e.g.,
```
public class AccountsCommitInterceptor implements
    CommitInterceptor<AccountID, MerkleAccount, AccountProperty> {

    // The tracker this interceptor should use for previewing changes. The interceptor is NOT
    // responsible for calling reset() on the tracker, as that will be done by the client code. 
    private final SideEffectsTracker sideEffectsTracker;
   
    @Override
    public void preview(List<AccountChanges<AccountID, MerkleAccount, AccountProperty>> changes) {
        // Here we compute the pending ℏ changes by comparing the result of
        // account.getBalance() with (long) changes.get(BALANCE); and then call 
        // sideEffectsTracker.trackHbarChange() to construct the record. If after all
        // balance changes are tracked, we see the sum is not net-zero, we need to
        // throw IllegalStateException.
        ...
    }
}
```